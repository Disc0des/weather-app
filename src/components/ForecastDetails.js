import React from "react";
import PropTypes from "prop-types";
import moment from "moment";

function ForecastDetails({ forecast }) {
  const { date, temperature, wind, humidity } = forecast;
  return (
    <div className="forecast-details">
      <div className="forecast-details_date" data-testid="forecast-date">
        {moment(date).format("ddd Do MMM")}
      </div>
      <div
        className="forecast-details_max-temperature"
        data-testid="forecast-temp-max"
      >
        Max temperature: {temperature.max} &deg;C
      </div>
      <div
        className="forecast-details_min-temperature"
        data-testid="forecast-temp-min"
      >
        Min temperature: {temperature.min} &deg;C
      </div>
      <div className="forecast-details_wind" data-testid="forecast-wind">
        Wind: {wind.speed}mph {wind.direction}
      </div>
      <div
        className="forecast-details_humidity"
        data-testid="forecast-humidity"
      >
        Humidity: {humidity}%
      </div>
    </div>
  );
}

export default ForecastDetails;
ForecastDetails.propTypes = {
  forecast: PropTypes.shape({
    date: PropTypes.number.isRequired,
    humidity: PropTypes.number.isRequired,
    temperature: PropTypes.shape({
      min: PropTypes.number,
      max: PropTypes.number,
    }).isRequired,
    wind: PropTypes.shape({
      speed: PropTypes.number,
      direction: PropTypes.string,
    }).isRequired,
  }).isRequired,
};
