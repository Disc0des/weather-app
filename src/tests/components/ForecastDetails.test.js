import React from "react";
import { render } from "@testing-library/react";
import ForecastDetails from "../../components/ForecastDetails";

describe("ForecastDetails", () => {
  const validProps = {
    date: 111111,
    humidity: 20,
    wind: {
      speed: 12,
      direction: "n",
    },
    temperature: {
      min: 12,
      max: 22,
    },
  };

  it("renders the correct date, temperature, wind and humidity props", () => {
    const { asFragment } = render(<ForecastDetails forecast={validProps} />);
    expect(asFragment()).toMatchSnapshot();
  });

  it("renders the correct values for props", () => {
    const { getByTestId } = render(<ForecastDetails forecast={validProps} />);
    expect(getByTestId("forecast-date")).toHaveClass("forecast-details_date");
    expect(getByTestId("forecast-humidity")).toHaveClass(
      "forecast-details_humidity"
    );
    expect(getByTestId("forecast-wind")).toHaveClass("forecast-details_wind");
    expect(getByTestId("forecast-temp-min")).toHaveClass(
      "forecast-details_min-temperature"
    );
    expect(getByTestId("forecast-temp-max")).toHaveClass(
      "forecast-details_max-temperature"
    );
  });
});
