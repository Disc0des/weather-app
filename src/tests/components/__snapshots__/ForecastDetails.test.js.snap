// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ForecastDetails renders the correct date, temperature, wind and humidity props 1`] = `
<DocumentFragment>
  <div
    class="forecast-details"
  >
    <div
      class="forecast-details_date"
      data-testid="forecast-date"
    >
      Thu 1st Jan
    </div>
    <div
      class="forecast-details_max-temperature"
      data-testid="forecast-temp-max"
    >
      Max: 22 °C
    </div>
    <div
      class="forecast-details_min-temperature"
      data-testid="forecast-temp-min"
    >
      Min: 12 °C
    </div>
    <div
      class="forecast-details_wind"
      data-testid="forecast-wind"
    >
      Wind: 12mph n
    </div>
    <div
      class="forecast-details_humidity"
      data-testid="forecast-humidity"
    >
      Humidity: 20%
    </div>
  </div>
</DocumentFragment>
`;
